src/export/FSHExporter.ts

```diff
@@ -63,150 +75,190 @@ export class FSHExporter {
-        exports.delete(key);
+    const writtenFiles: Map<string, string> = new Map();
+
+    const index: string[][] = [];
+    files.forEach((exportables, file) => {
+      // Aliases are each their own "exportable", but should be joined together
+      // by a single EOL, not double EOLs, and they should not be written in index.txt
+      // so they are handled separately
+      const [aliases, namedExportables] = partition(
+        exportables,
+        exportable => exportable instanceof ExportableAlias
+      );
+      const fileContent = [
+        aliases.map(a => a.toFSH()).join(EOL),
+        namedExportables.map(e => e.toFSH()).join(`${EOL}${EOL}`)
+      ]
+        .join(EOL)
+        .trim();
+      // Ignore empty files, and don't write them to index.txt
+      if (!fileContent) {
+        return;
       }
+      writtenFiles.set(file, fileContent);
+      namedExportables.forEach((exportable: NamedExportable) => {
+        // The index will have the name, FSH type, and file of the entity
+        index.push([exportable.name, exportable.constructor.name.replace('Exportable', ''), file]);
+      });
     });
 
-    return exports;
+    // Alphabetically sort the index by the name of the entity
+    index.sort((line1, line2) => (line1[0] > line2[0] ? 1 : -1));
+    index.unshift(['Name', 'Type', 'File']);
+    writtenFiles.set('index.txt', table(index));
+
+    return writtenFiles;
   }
 
-  private groupAsSingleFile(): Map<string, string> {
-    const results: string[] = [];
-    if (this.fshPackage.aliases.length > 0) {
-      results.push(this.fshPackage.aliases.map(a => a.toFSH()).join(EOL));
-    }
-    for (const profile of this.fshPackage.profiles) {
-      results.push(profile.toFSH());
-    }
-    for (const extension of this.fshPackage.extensions) {
-      results.push(extension.toFSH());
-    }
-    for (const codeSystem of this.fshPackage.codeSystems) {
-      results.push(codeSystem.toFSH());
-    }
-    for (const valueSet of this.fshPackage.valueSets) {
-      results.push(valueSet.toFSH());
-    }
-    for (const instance of this.fshPackage.instances) {
-      results.push(instance.toFSH());
-    }
-    for (const invariant of this.fshPackage.invariants) {
-      results.push(invariant.toFSH());
-    }
-    for (const mapping of this.fshPackage.mappings) {
-      results.push(mapping.toFSH());
-    }
-    return new Map().set('resources.fsh', results.join(`${EOL}${EOL}`));
+  private groupAsSingleFile(): Map<string, Exportable[]> {
+    const results: Exportable[] = [];
+    results.push(...this.fshPackage.aliases);
+    results.push(...this.fshPackage.profiles);
+    results.push(...this.fshPackage.extensions);
+    results.push(...this.fshPackage.codeSystems);
+    results.push(...this.fshPackage.valueSets);
+    results.push(...this.fshPackage.instances);
+    results.push(...this.fshPackage.invariants);
+    results.push(...this.fshPackage.mappings);
+    return new Map().set('resources.fsh', results);
```

Comment:
When comparing the single-file output between this and the previous version of this branch (and master), there is no additional line between the end of the Alias list and the first profile. Are we okay with that? It feels like there might not be a clean way to get around, in which case, I think this difference is probably okay.

Reply:
Oh that's kinda icky though. I will figure out a way to add a line.

Reply:
I'm glad this wasn't actually icky.